### PR TITLE
Availability: refactoring and tests

### PR DESCRIPTION
**Pull request by frootloops**

### CODE REVIEW

File: test/1_stdlib/Availability.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import SwiftShims
+
+let AvailabilityTests = TestSuite("Availability")
+AvailabilityTests.test("==, <, >=") {
```
Comment (gribozavr) [inline test/1_stdlib/Availability.swift]: We have tests for this API in `test/1_stdlib/Runtime.swift` (`AvailabilityVersionsTestSuite`).

Could you check if any of these cases are not covered by those tests, and merge them if so?

File: stdlib/public/core/Availability.swift [hunk 2]
```diff
@@ -61,31 +61,15 @@ public func < (
...
-  if lhs.patchVersion > rhs.patchVersion {
-    return false
-  }
-
-  if lhs.patchVersion < rhs.patchVersion {
-    return true
-  }
-
-  return false
+  return lhs.patchVersion < rhs.patchVersion
```
Comment (gribozavr) [inline stdlib/public/core/Availability.swift]: Very good simplification!

**Comment by frootloops** [comment]
@gribozavr it looks like all tests already there.

**Comment by frootloops** [comment]
@gribozavr I want to add some value (add a new function maybe) for Swift users but I don't know how can I find what exactly I should add. Maybe you're going to add something that you can entrust me? Or you could give me a direction or an advice about what I can add.

Thank you for advice :)

**Comment by gribozavr** [comment]
@frootloops Try one of these:

https://bugs.swift.org/browse/SR-306
https://bugs.swift.org/browse/SR-368

Note that adding new APIs requires following the evolution process: https://github.com/apple/swift-evolution

You can work on the implementation and tests more or less separately from the API design.  Just start with some reasonable first approximation for the API.  I am confident that it would be easy to adapt the implementation to any reasonable API design that swift-evolution settles on, especially for the stable sort function.

**Pull request merged**